pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -160,15 +168,28 @@ def __init__(self, mps, wires, id=None):
         assert qml.math.isclose(
             new_dj0, dj2
         ), "Dimension mismatch: the last tensor's first dimension does not match the previous third dimension."
-        super().__init__(*mps, wires=wires, id=id)
+
+        self.hyperparameters["input_wires"] = qml.wires.Wires(wires)
+
+        if work_wires:
+            self.hyperparameters["work_wires"] = qml.wires.Wires(work_wires)
```

Comment:
Do we require a check that sufficient work wires have been provided?

Reply:
I just added the check in the decomposition 👍
(in case of lightning.tensor they are not required)

Reply:
If work_wires are provided and the user uses lightning.tensor, is there any issue?

Reply:
Good point :) They don't run anything on their side. They just use this template as a container to extract `MPSPrep.mps` so we are safe here